pkg/minikube/tunnel/route_linux.go

```diff
@@ -54,7 +54,7 @@ func (router *osRouter) EnsureRouteIsAdded(route *Route) error {
 }
 
 func (router *osRouter) Inspect(route *Route) (exists bool, conflict string, overlaps []string, err error) {
-	cmd := exec.Command("netstat", "-nr", "-f", "inet")
+	cmd := exec.Command("route", "-n")
```

Comment:
Oh, great find! The man page states that: 

```
 to SELECTOR (default)
                     only select routes from the given range of destinations.  SELECTOR consists of an optional modifier (root, match or exact) and a prefix.  root PREFIX selects routes with prefixes not shorter than PREFIX.  F.e.  root 0/0 selects the entire routing table.
                     match PREFIX selects routes with prefixes not longer than PREFIX.  F.e.  match 10.0/16 selects 10.0/16, 10/8 and 0/0, but it does not select 10.1/16 and 10.0.0/24.  And exact PREFIX (or just PREFIX) selects routes with this exact prefix. If neither of
                     these options are present, ip assumes root 0/0 i.e. it lists the entire table.
```

So for the tunnel's "show me all intersecting destination ranges" query I'll need to run both with `match` and `root`, so that I can get routes that have destination ranges that are either shorter or longer and I'll _still_ need to filter out "default via ..." type of rows and merge the two results.

E.g: 
I created a bogus but overlapping route: `10.110.0.0/16 - > 127.0.0.1` - it only shows up in `root` as it's longer than my inspected 12 bit range: 

```
$ ip route show to match 10.96/12
default via 172.31.126.254 dev eno1 proto dhcp metric 100 
10.96.0.0/12 via 192.168.39.47 dev virbr1

$ ip route show to root 10.96/12
10.96.0.0/12 via 192.168.39.47 dev virbr1 
10.110.0.0/16 via 127.0.0.1 dev lo 
```

Thus, I still have to parse + on top of it I'd have to merge the two command executions + will have a bit of deviation from the other OSes. 
Not sure it's worth saving some extra parsing. 
If only I could combine `root` and `match` ...

Reply:
Bummer :( Parsing it is!